src/shared/ScoreProvider.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useContext, useState } from 'react';
+
+export enum ScoreAction {
+  INCREMENT,
+  DECREMENT
+}
+
+export type ScoreStore = Readonly<{
+  score: number;
+  updateScore(action: ScoreAction): void;
+}>;
+
+const ScoreContext = React.createContext<ScoreStore>(null);
+export const ScoreProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [ score, setScore ] = useState(0);
+
+  const increment = () => {
+    setScore(score + 1);
```

Comment:
Consider setting score as setScore((currScore) => currScore + 1). There's a bug that can happen if setScore is called multiple times in a short period where score isn't updated properly between calls so the number would be smaller than it should be.

See https://youtu.be/0yzoAbrjV6k?t=246 for a more thorough explanation.

Reply:
Nice one. I didn't know this! Thanks.